### DISCUSSION TITLE
What is the correct Serial Port for CC2652R?

### DISCUSSION THREAD

**Discussion by NimKat16** [Koenkk/zigbee2mqtt#9027] [General]
Hello,

I am very new to this so please bear with me :)

I am trying to connect Slaesh CC2652R to HomeAssitant. It is supposed to be flashed with zigbee2mqtt firmware (can I actually verify this...?)

I use a Windows10 PC and run a Oracle VM VirtualBox to get to HA. When I try to start zigbee2mqtt addon, i see an error starting the herdsman stating that my serial port address does not exist.
![image](https://user-images.githubusercontent.com/91806336/136102724-57439704-ecb1-460e-bbfb-783e04ad8183.png)

My stick is connected to a USB port and is assigned COM3.
I try to see what is the serial port in Supervisor > System >Host > Hardware but i see nothing remotely resembling the stick. I had a few events and ttySx (x = 0..3).
I try ttySx but i get an error on baud rate
Zigbee2MQTT:error 2021-10-05 23:17:31: Error: Error while opening serialport 'Error: Error: I/O error setting custom baud rate of 115200'

I installed the VM Extension Pack to support USBs. I also enabled Serial Ports in VM and restarted everything. Now i see a lot of things in Supervisor > System >Host > Hardware. But nothing resembling the stick :(
![image](https://user-images.githubusercontent.com/91806336/136108339-874c3bde-ac9f-4f5a-926a-be12134bcba7.png)

Anyone happens to know the reason I cannot see the stick in the list of available hardware?

Ciao,
Nima

**Comment by NimKat16** [comment]
No one has any idea on what iss wrong? I still cannot manage to 'see' my CC2652R in HA :(

**Comment by francisp2** [comment]
Did you enable the USB filter ? 

![image](https://user-images.githubusercontent.com/52328611/137087616-e1c0abe2-dd11-4a8a-857c-ab71cfe3b757.png)

**Reply by NimKat16** [reply]
Thanks for the reply! I had not done that thinking filtering is something to limit the selection... i was obvious wrong... So now i did enable the filter as shown below.. but i still do not get my device in Hardware :(
![image](https://user-images.githubusercontent.com/91806336/137119948-b6c58483-5183-40d2-ab9d-033aea1cbc1b.png)

I see people stating that 'lsusb' or 'dmesg | grep 'tty' shows the serial ports accessible but that does not work in HA command line i have in my VM, it supports very few commands; i guess somehow they have HA on LINUX not in a VM. Soin case the Supervisor > System >Host > Hardware is the 'truth'; i still do not see my CC2652R :(

**Comment by NimKat16** [comment]
OK,this seems to be solved by enabling USB filter (thanks Francisp2) for both UART and stick + making sure that I actually connect the stick AFTER HA has been started up... 

I could not even see the "Silicon Labs .." in the filters... After HA is started, then i can see it and add it to the list of filters. But even after adding it, it works only, as i said, if the stick is connected AFTER HA has been started up... weird!!

![image](https://user-images.githubusercontent.com/91806336/137600793-fe7099a5-7565-4b34-8669-1841791f4160.png)
